dovpanda/base.py

```diff
@@ -90,6 +90,25 @@ def tell(self, *args, **kwargs):
     def set_output(self, output):
         self.teller.set_output(output)
 
+    def attach_hooks(self, f, func_hooks):
```

Comment:
Wasn't the point of this was that now it can work only with self?
f is the keys of self.hooks while the values are (func_hooks, hook_type)

Reply:
@galbraun

Reply:
Yep, you are right. fixed.